flight/src/types.rs

```diff
@@ -0,0 +1,102 @@
+use std::fmt;
+
+use cid::Cid;
+use serde::{Deserialize, Serialize};
+use serde_bytes::ByteBuf;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum ConclusionEvent {
+    Data(ConclusionData),
+    Time(TimeEvent),
+}
+
+// TODO_Discussion: Do we still support all these types?
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub enum StreamType {
```

Comment:
We have a type like this already in `ceramic-core`. Can we re_use the same type?

Reply:
I thought the goal was that the core protocol didn't need to know anything about streamtypes.  Can the flight API just pass through the streamtype as an integer without knowing anything about what it represents?  The aggregator will of course need to be able to interpret the streamtype and know what it means, but does the conclusion feed need that level of knowledge?

As written now, if C1 started seeing events with a new streamtype, would the conclusion feed break?  I don't think we want that, it should be possible to use an unmodified C1 with a custom aggregator that supports a new custom streamtype without needing to update C1.

Reply:
Yeah the custom feed would break in this code with a Unknown Stream Type error. We can make the Conclusion Feed unaware of stream types and have it just store whatever stream type it gets in the database? But this means we might be storing arbitrary stream types. Is that what we want?

Reply:
yeah this is a good question.  If a user writes a custom aggregator with a new streamtype and starts using it, but neglects to update the streamtype table to reserve the id number for their new streamtype, what should happen?  It would be kind of nice if C1 enforced that every event belongs to *some* registered streamtype, without caring which.  I could imagine in the future C1 loading the streamtype list at startup and rejecting events with unregistred streamtypes.  So maybe for now we should just have a hardcoded list of acceptable ID numbers and enforce that the streamtype we see in the allowed set?
But still without parsing or interpreting which specific streamtype id is used, just validating that it is one of the valid ones.

WDYT @nathanielc?  Open to pushback or other ideas here.

Reply:
I think this code doesn't care which stream type it is and just need to pass that information along. If we want to validate stream types on ingest we can do that as part of the ceramic-service validation logic. 

The conclusion feed should treat it as an opaque integer

Reply:
Yeah, agreed in the context of this PR nothing more to do, I was more trying to ask the question in general of how we think this should be handled.  If we think it's a good idea to validate that incoming events are part of a known streamtype, I can add a ticket to the Event Validation project to do that.